File: problemset2/readability/readability.c
#include "../../cs50.h"
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void) {
    string text = get_string("Text: ");

    int letters = 0;
    int words = 0;
    int sentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++) {
        char character = text[i];
        if (islower(text[i]) || isupper(text[i])) {
            letters++;
        } else {
            switch (text[i]) {
                case 32:
                    words++;
                    break;
                case 33:
                case 46:
                case 63:
                    sentences++;
                    break;
                default:
                    break;
            }
        }
    }
    words++;

    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8;

    if (index > 16) {
        printf("Grade 16+\n");
    } else if (index < 0) {
        printf("Before Grade 1\n");
    } else {
        printf("Grade %i\n", (int) round(index));
    }
}

File: problemset6/sentimental-readability/readability.py
from cs50 import get_string
import re

text = get_string("Text: ")

words_result = re.findall(" ", text)
words_n = len(words_result) + 1

sentences_result = re.findall("[!.?]", text)
sentences_n = len(sentences_result)

letters_result = re.findall("[A-Za-z]", text)
letters_n = len(letters_result)

letters_per_hundred_words = (letters_n / words_n) * 100
sentences_per_hundred_words = (sentences_n / words_n) * 100

index = round(
    0.0588 * letters_per_hundred_words - 0.296 * sentences_per_hundred_words - 15.8
)
if index < 1:
    print(f" Before Grade 1")
elif index >= 16:
    print(f"Grade 16+")
else:
    print(f"Grade {index}")
